infrastructure/terraform/modules/templates-api/module_get_template_lambda.tf

```diff
@@ -0,0 +1,52 @@
+module "get_template_lambda" {
+  depends_on  = [module.build_template_lambda, module.build_template_client]
```

Comment:
you shouldn't need depends_on for resources that are explicitly referenced in the resource

Reply:
Oh i didn't know that! I'll try it out!

Reply:
Ah, so I just tried removing the depends_on but I got a bunch of errors. Might be one to talk through tomorrow!

Reply:
Terraform tells me I need to use `depends_on` others a `circular dependency` error is thrown by Terraform.  

I want to build the `backend-api` and `backend-client` once rather than building multiple times per lambda.